lib/ProductOpener/Display.pm

```diff
@@ -3849,6 +3849,17 @@ sub search_and_display_products($$$$$) {
 			products => [],
 		};
 
+		# Fields displayed in HTML results pages
```

Comment:
We are going to cache the results of the Mongo query. After we get the results back, we save them in memcached like that:

$memd->set($key, $request_ref->{structured_response}, 3600) or $log->debug("Could not set value for MongoDB query key", { key => $key }); 

So if we limit the fields returned, we need to make the key specific to those fields.

Currently the key is set above:
my $mongodb_query_ref = [ lc => $lc, query => $query_ref, sort => $sort_ref, limit => $limit, skip => $skip ];
my $key = $server_domain . "/" . freeze($mongodb_query_ref);
$key = md5_hex($key);

So we need to move the conditions for limiting the fields  (   if ($request_ref->{json} or $request_ref->{jsonp} or $request_ref->{xml} or $request_ref->{jqm} or $request_ref->{rss}) {   )  above, before we generate the key.

Reply:
Ok I got it. I will correct that.